lib/builder.dart

```diff
@@ -28,14 +43,14 @@ class _VersionBuilder implements Builder {
     }
 
     await buildStep.writeAsString(
-        AssetId(buildStep.inputId.package, 'lib/src/version.dart'), '''
-// Generated code. Do not modify.
-const packageVersion = '${pubspec.version}';
+        AssetId(buildStep.inputId.package, 'lib/$targetPath'), '''
+/// Generated code. Do not modify.
+const String $fieldName = '${pubspec.version}';
```

Comment:
To avoid linter warnings : `public_member_api_docs` and `type_annotate_public_apis`.

Reply:
We should not expect generated code to follow all lints.

I personally consider enforcing these lints an anti-pattern. Changing a block comment arbitrarily into a doc comment to satisfy a lint is also a very bad idea.

Reply:
> We should not expect generated code to follow all lints.

Should these lints be muted in the generated code with `ignore_for_file`, so that they don't come up with the analyzer?

Reply:
They could be. Better yet would be to be able to ignore _lints_ from these files without ignoring all analysis.

https://github.com/dart-lang/sdk/issues/34069